### PR TITLE
[SPARK-34498][SQL][TESTS]  fix the remaining problems in  #31560

### PR DESCRIPTION
**Pull request by kevincmchen**
### What changes were proposed in this pull request?

This is a followup of #31560,  
In  #31560,  we added `JavaSimpleWritableDataSource ` and left some little problems like unused interface `SessionConfigSupport` 、 inconsistent schema between `JavaSimpleWritableDataSource ` and `SimpleWritableDataSource`.
This PR fixes the remaining problems in #31560.

### Why are the changes needed?

1. `SessionConfigSupport` in `JavaSimpleWritableDataSource ` and `SimpleWritableDataSource` is never used, so we don't need to implement it.
2. change the schema of `SimpleWritableDataSource`, to match `TestingV2Source`

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

existing testsuites

### CODE REVIEW

**Comment by srowen** [comment]
Jenkins test this please

File: sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala [hunk 1]
```diff
@@ -41,11 +41,9 @@ import org.apache.spark.util.SerializableConfiguration
  * Each task writes data to `target/_temporary/uniqueId/$jobId-$partitionId-$attemptNumber`.
  * Each job moves files from `target/_temporary/uniqueId/` to `target`.
  */
-class SimpleWritableDataSource extends SimpleTableProvider with SessionConfigSupport {
+class SimpleWritableDataSource extends SimpleTableProvider {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala:44]: can we extend `TestingV2Source` instead?

Comment (kevincmchen) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala:44]: yes

File: sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala [hunk 2]
```diff
@@ -222,7 +217,7 @@ class CSVDataWriter(fs: FileSystem, file: Path) extends DataWriter[InternalRow]
   private val out = fs.create(file)
 
   override def write(record: InternalRow): Unit = {
-    out.writeBytes(s"${record.getLong(0)},${record.getLong(1)}\n")
+    out.writeBytes(s"${record.getInt(0)},${record.getInt(1)}\n")
```
Comment (srowen) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala:220]: This change just matches the TestingV2Source.schema?

**Review by srowen — COMMENTED** [review]
Looks OK

File: sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala [hunk 3]
```diff
@@ -68,7 +63,7 @@ class SimpleWritableDataSource extends SimpleTableProvider with SessionConfigSup
       new CSVReaderFactory(serializableConf)
     }
 
-    override def readSchema(): StructType = tableSchema
+    override def readSchema(): StructType = TestingV2Source.schema
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala:66]: `SimpleScanBuilder.readSchema` is already `TestingV2Source.schema`, we don't need to override it here.

File: sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala [hunk 4]
```diff
@@ -134,7 +129,7 @@ class SimpleWritableDataSource extends SimpleTableProvider with SessionConfigSup
     private val path = options.get("path")
     private val conf = SparkContext.getActive.get.hadoopConfiguration
 
-    override def schema(): StructType = tableSchema
+    override def schema(): StructType = TestingV2Source.schema
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala:132]: ditto, `SimpleBatchTable.schema` is already `TestingV2Source.schema`

File: sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala (hunk 2 above)
Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala:220]: Yea

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala (hunk 3 above)
Comment (kevincmchen) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala:66]: yes, it‘s unnecessary to override the function.

File: sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala (hunk 4 above)
Comment (kevincmchen) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/SimpleWritableDataSource.scala:132]: yes, it‘s unnecessary.

**Comment by srowen** [comment]
Jenkins test this please

**Comment by cloud-fan** [comment]
GA passed, merging to master, thanks!

**Pull request closed without merging**